internal/cache/metadata/stat_cache_test.go

```diff
@@ -511,7 +511,8 @@ func (t *StatCacheTest) Test_ShouldReturnHitFalseWhenOnlyObjectAlreadyHasEntry()
 
 	hit, entry := t.statCache.LookUpFolder(name, someTime)
 
-	assert.False(t.T(), hit)
+	// If "key1" object exist then corresponding folder entry will be nil, but hit will be true as key have entry in the cache for object.
+	assert.True(t.T(), hit)
 	assert.Nil(t.T(), entry)
```

Comment:
if folder is nil, isn't that a problem: https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/fs/inode/dir.go#L387

Reply:
We are returning NotFound error in that case. [ref](https://github.com/GoogleCloudPlatform/gcsfuse/blob/9c0ff1dff36a1de9c0b9b11dfaa47a495de84359/internal/storage/caching/fast_stat_bucket.go#L378). Which means it should hit [this](https://github.com/GoogleCloudPlatform/gcsfuse/blob/v2.5.0/internal/fs/inode/dir.go#L382) case and return nil,nil.